### PR TITLE
ATO-369: Add stack for deploying TxMA infrastructure to Orchestration accounts

### PR DESCRIPTION
**Pull request by billrdunn**
## What

- TxMA infrastructure added as a manual stack - it is intended to be deployed to Orchestration accounts only once, and not as part of the pipeline.
- Note that the stack has already been deployed to all envs.

## How to review

1. Code review

## 🚨⚠️ Orchestration and Authentication mutual dependencies ⚠️ 🚨

Be careful when making changes to code in 'shared' components where each team has a copy.

- [x] Impact on orch and auth mutual dependencies has been checked.

### CODE REVIEW

File: ci/stack-orchestration/manual-stacks/txma/README.md [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+
+### Updating the Stack
+
+Set your AWS profile to the correct environment, run `export ENVIRONMENT=<environment>` then run:
+
+```bash
+aws cloudformation update-stack --stack-name txma \
+  --template-body file://$(pwd)/template.yml \
+  --region eu-west-2 \
+  --capabilities CAPABILITY_NAMED_IAM CAPABILITY_AUTO_EXPAND \
```
Comment (ethanmills) [inline ci/stack-orchestration/manual-stacks/txma/README.md:39]: Do we actually need these capabilities? I could believe we might need NAMED_IAM, but I'd be surprised if we need AUTO_EXPAND

File: ci/stack-orchestration/manual-stacks/txma/template.yml [hunk 2]
```diff
@@ -0,0 +1,203 @@
...
+  TxMASQSProducerAuditEventQueue:
+    Type: AWS::SQS::Queue
+    Properties:
+      MessageRetentionPeriod: 1209600 # 14 days in seconds
+      QueueName: !Sub '${AWS::StackName}-AuditEventQueue'
+      KmsMasterKeyId:
+        Fn::If:
+          - TxMAIntegrationEnabled
+          - !Ref TxMASQSProducerAuditEventQueueEncryptionKeyAlias
+          - !Ref AWS::NoValue
```
Comment (ethanmills) [inline ci/stack-orchestration/manual-stacks/txma/template.yml:43]: Why this distinction?

**Review by ethanmills — COMMENTED** [review]
When and where do you think we should implement the alerting? In a followup? In this stack or another?

File: ci/stack-orchestration/manual-stacks/txma/README.md (hunk 1 above)
Comment (billrdunn) [reply ci/stack-orchestration/manual-stacks/txma/README.md:39]: Neither needed

File: ci/stack-orchestration/manual-stacks/txma/template.yml (hunk 2 above)
Comment (billrdunn) [reply ci/stack-orchestration/manual-stacks/txma/template.yml:43]: If `TxMAIntegrationEnabled = false` then `TxMASQSProducerAuditEventQueueEncryptionKeyAlias` isn't provisioned so can't be assigned to `KmsMasterKeyId`. Is that what you meant?

File: ci/stack-orchestration/manual-stacks/txma/template.yml [hunk 3]
```diff
@@ -0,0 +1,201 @@
...
+          - Sid: 'Allow Lambda Access'
+            Effect: Allow
+            Principal:
+              Service: 'lambda.amazonaws.com'
+            Action:
+              - kms:Encrypt
+              - kms:Decrypt
+              - kms:ReEncrypt
+              - kms:GenerateDataKey
+            Resource: '*'
```
Comment (ethanmills) [inline ci/stack-orchestration/manual-stacks/txma/template.yml:114]: What is this statement for?

File: ci/stack-orchestration/manual-stacks/txma/template.yml (hunk 2 above)
Comment (billrdunn) [reply ci/stack-orchestration/manual-stacks/txma/template.yml]: Now conditionally provisioning only resources related to decryption of events by TxMA

File: ci/stack-orchestration/manual-stacks/txma/template.yml (hunk 3 above)
Comment (billrdunn) [reply ci/stack-orchestration/manual-stacks/txma/template.yml:114]: Nothing, right? 
The previous statement allows the root account to perform KMS actions on all KMS resources. This statement allows all lambdas to perform the listed actions on all KMS resources. Therefore this statement is redundant (and potentially insecure as it allows all lambdas across AWS, not just those in the root account).

**Review by ethanmills — APPROVED** [review]
Thanks!

File: ci/stack-orchestration/manual-stacks/txma/template.yml (hunk 3 above)
Comment (andyloughran) [reply ci/stack-orchestration/manual-stacks/txma/template.yml]: Good spot, glad that this got picked up.  Well done.

**Pull request merged**